tom/bot.py

```diff
@@ -2,34 +2,38 @@
 import random
 import json
 import logging as log
+from copy import copy
 
 from tom.github import GitHub, GitHubInterface, PR
 from tom.jenkins import Jenkins
 from tom.slack import CommandDispatcher
-from tom.utils import confirmation
-
-
-class Tom():
-    def __init__(self, secrets, interactive):
-        github = secrets["GITHUB_TOKEN"]
-        self.github = GitHub(github)
-
-        user = secrets["JENKINS_USER"]
-        token = secrets["JENKINS_TOKEN"]
-        if not (user and token):
-            sys.exit("Cannot start Tom without Jenkins credentials")
-        crumb = secrets["JENKINS_CRUMB"]
-        self.jenkins = Jenkins(user, token, crumb)
+from tom.utils import confirmation, pretty
+
+
+class Bot():
+    def __init__(
+            self, username, secrets, orgs, repo_maintainers, jenkins, job, maintainers, trusted,
+            directory, interactive):
```

Comment:
I think this should be a dictionary/namedtuple or something like that. Not a bunch of so many arguments.